react/admin/OrganizationDetails.tsx

```diff
@@ -259,7 +259,7 @@ const OrganizationDetails: FunctionComponent = () => {
   useEffect(() => {
     const customFieldsToShow = joinById(
       data?.getOrganizationById?.customFields || [],
-      defaultCustomFieldsData?.getB2BSettings.organizationCustomFields || []
```

Comment:
I think it would be better to handle this in the backend. Regardless of what is in Vbase you can return a default value in the GraphQL response so the frontend doesn't need to have special handling for the edge case.

Reply:
What if `defaultCustomFieldsData` were not setup?

Reply:
What do you mean by "not setup"?

Reply:
When someone just added the `b2b-organizations` app to a project and haven't added the fields yet, `defaultCustomFieldsData` would come as null

Reply:
I've also added a check for customFieldsState to have items before showing a `<PageBlock>` to prevent this header from showing up if fields are not setup or not loaded yet: 
![image](https://user-images.githubusercontent.com/44775362/192842638-b7369e69-695d-448f-b7a1-fae10a737358.png)

Reply:
The scenario where the app is newly installed is what I was talking about in the original comment. Just because the data doesn't exist in Vbase yet doesn't mean it can't be populated in the API, we can add a default value in the resolver. In general we should try to keep data consistent throughout as much of the system as possible and hide implementation details like whether we have a value in Vbase, the frontend doesn't need to know or care about that.

If we want to to hide the header we do need logic on the frontend though, and I didn't think about the case where it is loading, that also requires handling on the frontend so I think this is fine.

Reply:
I've made a change in b2b-organizations-graphql app so that if `organizationCustomFields` is null, we return empty an array